tasks/steps/blocks/extend_systemd_unit_files.yml

```diff
@@ -0,0 +1,31 @@
+---
+
+- name: 'Create directory for systemd unit files extensions'
+  file:
+    state: '{{ "directory" if cartridge_extra_env else "absent" }}'
+    recurse: '{{ true if cartridge_extra_env else false }}'
+    path: '{{ (cartridge_systemd_dir, instance_info.systemd_service_dir) | cartridge_path_join }}'
```

Comment:
It's `systemd_service_dir_name` if you need to join it. BTW, why don't you compute abspath on getting instance info?